internal/providers/docker/components/container/lifecycle.go

```diff
@@ -165,14 +187,41 @@ func (c *Container) create(ctx context.Context) error {
 		//Init *bool `json:",omitempty"`
 	}
 	for _, mapping := range c.Spec.Ports {
-		target := nat.Port(mapping.Target) // TODO: Handle ranges.
-		bindings := hostCfg.PortBindings[target]
-		bindings = append(bindings, nat.PortBinding{
-			HostIP:   mapping.HostIP,
-			HostPort: mapping.Published,
-		})
-		// TODO: Handle mapping.Mode and mapping.Protocol.
-		hostCfg.PortBindings[target] = bindings
+		target, err := nat.NewPort(mapping.Protocol, mapping.Target)
+		if err != nil {
+			return fmt.Errorf("could not parse port: %w", err)
+		}
+
+		low, high, err := target.Range()
+		if err != nil {
+			return fmt.Errorf("could not parse port range: %w", err)
+		}
+
+		for i := low; i <= high; i += 1 {
+			publishedPort, err := nat.NewPort(mapping.Protocol, mapping.Published)
+			if err != nil {
+				return fmt.Errorf("could not parse port range: %w", err)
+			}
+
+			publishedLow, publishedHigh, err := publishedPort.Range()
+			if err != nil {
+				return fmt.Errorf("could not parse port range: %w", err)
+			}
+
+			hostPort := publishedPort.Port()
+			if publishedHigh != publishedLow {
```

Comment:
Not sure I quite understand this bit. Also, I don't see any check to validate that the published/host ranges have the same size/length/number-of-ports... They are parallel arrays, right?

Reply:
Yes you're right — it didn't break when it should have. I've added some more validation now.